build.rs

```diff
@@ -28,7 +28,5 @@ fn main() -> Result<()> {
         std::process::exit(1);
     }
 
-    let mut config = Config::default();
-    *config.git_mut().sha_kind_mut() = ShaKind::Short;
-    vergen(config)
+    EmitBuilder::builder().git_sha(true).emit()
```

Comment:
Up above we've got the 'cargo:rerun-if-changed' lines, which is going to make this not re-run unless package.json or package-lock.json are changed, which isn't ideal.

Reply:
Also, the docs say there should be two colons between 'cargo' and 'rerun-if-changed', so now I'm wondering if any of this works (maybe a single colon works too).

Reply:
I'm looking at the docs and it only expects a single colon. https://docs.rs/vergen/latest/vergen/index.html

Reply:
The key there is that vergen itself emits these, which will override whatever we have in place for the typescript build. That's good.

Reply:
Assuming it doesn't *add* to them, in which case it would start making typescript stuff re-run unnecessarily ...

Reply:
Regarding one colon vs two: https://doc.rust-lang.org/cargo/reference/build-scripts.html
> Note: The old invocation prefix cargo: (one colon only) is deprecated and won’t get any new features. To migrate, use two-colons prefix cargo::, which was added in Rust 1.77. If you were using cargo:KEY=VALUE for arbitrary links manifest key-value pairs, it is encouraged to switch to cargo::metadata=KEY=VALUE. Stick to cargo: only if the support of Rust version older than 1.77 is required.

Reply:
Based on the vergen documentation, those values do seem to be additive. However, since vergen is only triggering on some git files, this is probably not going to cause excessive re-building of typescript.

Reply:
We've fixed the rust version to 1.73. https://github.com/aws/aperf/blob/main/.github/workflows/ci.yml#L32. The double colon fails in 1.73 saying it's reserved for future use and if we want it, we'll have to upgrade the rust version.